src/wiring_diagrams/Directed.jl

```diff
@@ -205,6 +205,12 @@ mutable struct WiringDiagram{Theory, PortValue, WireValue, BoxValue} <: Abstract
     # Copy constructor for shallow copy
     new{T, PortValue, WireValue, BoxValue}(copy(f.diagram), f.value)
   end
+  function WiringDiagram{T, PortValue, WireValue, BoxValue}(
```

Comment:
I know you spent a lot of time tracking down this bug, we don't actually need this new constructor, do we? If not, I'd suggest just removing it.

Reply:
Sorry, I guess I was wrong: the test failures suggest that you did need this constructor after all.